regresssion in 3.5.28 with new validation code

we're getting this NPE in fabric8-platform in `apps/ingress-nginx` when running `mvn fabric8:resource`:
```
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.28:resource (default-cli) on project ingress-nginx: Execution default-cli of goal io.fabric8:fabric8-maven-plugin:3.5.28:resource failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.28:resource (default-cli) on project ingress-nginx: Execution default-cli of goal io.fabric8:fabric8-maven-plugin:3.5.28:resource failed.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal io.fabric8:fabric8-maven-plugin:3.5.28:resource failed.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 20 more
Caused by: java.lang.NullPointerException
	at io.fabric8.maven.core.util.validator.ResourceValidator.geFileContent(ResourceValidator.java:169)
	at io.fabric8.maven.core.util.validator.ResourceValidator.validate(ResourceValidator.java:108)
	at io.fabric8.maven.plugin.mojo.build.ResourceMojo.validateIfRequired(ResourceMojo.java:286)
	at io.fabric8.maven.plugin.mojo.build.ResourceMojo.executeInternal(ResourceMojo.java:270)
	at io.fabric8.maven.plugin.mojo.AbstractFabric8Mojo.execute(AbstractFabric8Mojo.java:74)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
```

A PR is on its way...

--- a/core/src/main/java/io/fabric8/maven/core/util/validator/ResourceValidator.java
+++ b/core/src/main/java/io/fabric8/maven/core/util/validator/ResourceValidator.java
@@ -22,7 +22,6 @@ import com.networknt.schema.JsonSchema;
 import com.networknt.schema.JsonSchemaFactory;
 import com.networknt.schema.ValidationMessage;
 import io.fabric8.maven.core.util.ResourceClassifier;
-import io.fabric8.maven.core.util.validator.IgnoreRule;
 import io.fabric8.maven.docker.util.Logger;
 
 import javax.validation.ConstraintViolation;
@@ -103,13 +102,15 @@ public class ResourceValidator {
      * @throws IOException
      */
     public int validate() throws ConstraintViolationException, IOException {
-        for(File resource: resources) {
-            log.info("validating %s resource", resource.toString());
-            JsonNode resourceNode = geFileContent(resource);
-            JsonSchema schema = getJsonSchema(prepareSchemaUrl(resourceNode));
+        for (File resource : resources) {
+            if (resource.isFile() && resource.exists()) {
+                log.info("validating %s resource", resource.toString());
+                JsonNode resourceNode = geFileContent(resource);
+                JsonSchema schema = getJsonSchema(prepareSchemaUrl(resourceNode));
 
-            Set<ValidationMessage> errors = schema.validate(resourceNode);
-            processErrors(errors, resource);
+                Set<ValidationMessage> errors = schema.validate(resourceNode);
+                processErrors(errors, resource);
+            }
         }
 
         return resources.length;
@@ -160,13 +161,9 @@ public class ResourceValidator {
     }
 
     private JsonNode geFileContent(File file) throws IOException {
-        InputStream resourceStream = null;
-        try {
-            resourceStream = new FileInputStream(file);
+        try (InputStream resourceStream = new FileInputStream(file)) {
             ObjectMapper jsonMapper = new ObjectMapper(new YAMLFactory());
             return jsonMapper.readTree(resourceStream);
-        } finally {
-            resourceStream.close();
         }
     }
 

--- a/plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java
+++ b/plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java
@@ -285,12 +285,6 @@ public class ResourceMojo extends AbstractResourceMojo {
             if (!skipResourceValidation) {
                 new ResourceValidator(resourceDir, classifier, log).validate();
             }
-        } catch (IOException e) {
-            if (failOnValidationError) {
-                throw new MojoExecutionException("Failed to validate resources", e);
-            } else {
-                log.warn("Failed to validate resources: %s", e.getMessage());
-            }
         } catch (ConstraintViolationException e) {
             if (failOnValidationError) {
                 log.error("[[R]]" + e.getMessage() + "[[R]]");
@@ -298,6 +292,12 @@ public class ResourceMojo extends AbstractResourceMojo {
                 throw new MojoFailureException("Failed to generate fabric8 descriptor");
             } else {
                 log.warn("[[Y]]" + e.getMessage() + "[[Y]]");
+            }
+        } catch (Throwable e) {
+            if (failOnValidationError) {
+                throw new MojoExecutionException("Failed to validate resources", e);
+            } else {
+                log.warn("Failed to validate resources: %s", e.getMessage());
             }
         }
     }

